inc/Engine/CriticalPath/APIClient.php

```diff
@@ -0,0 +1,234 @@
+<?php
+namespace WP_Rocket\Engine\CriticalPath;
+
+use WP_Error;
+
+/**
+ * Class APIClient
+ *
+ * @package WP_Rocket\Engine\CriticalPath
+ */
+class APIClient {
+
+	/**
+	 * Constant url for Critical Path API job.
+	 */
+	const API_URL = 'https://cpcss.wp-rocket.me/api/job/';
+
+	/**
+	 * Sends a generation request to the Critical Path API
+	 *
+	 * @since 3.6
+	 *
+	 * @param string $post_url The post URL.
```

Comment:
```suggestion
	 * @param string $url The URL to send a CPCSS generation request for.
```

The URL can be a post URL, but it can also be any kind of URL on a site, so it's better to be generic